Deprecate StarLine engine switch attributes

Anonym-tsk opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->

`ignition` and `autostart` state attributes of the engine switch are deprecated and will be removed in 2025.8. 2 new binary sensors are introduced to replace them.

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
- Binary sensor `running state` renamed to `ignition`.
- Added `autostart` binary sensor.
- `ignition` and `autostart` attributes of engine switch are deprecated.



## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [x] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->
According to [StarLine documentation](https://developer.starline.ru/#/User/UserInfoV2), `run` field contains `ignition state`, not `is car running`.
It was my mistake https://github.com/home-assistant/core/pull/130444

![image](https://github.com/user-attachments/assets/b31712dd-6953-4c96-8918-1dadbbefc671)


- This PR fixes or closes issue: fixes #
- This PR is related to issue: https://github.com/home-assistant/core/pull/133566
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/36540

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [x] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [x] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

Anonym-tsk committed:
Rename run sensor

Anonym-tsk committed:
Deprecated ignition attribute

Anonym-tsk committed:
Disabled by default

davet2001 reviewed (dismissed):
@Anonym-tsk Please could you help provide a bit more context for this change?

`is_running` is being replaced by `ignition`, but also deprecated?

Changing the translation key will add some work for the translation teams so I am not clear why this is a good idea if it is being deprecated.

For the deprecation warning, I couldn't find anything obvious. Possibly you could sublcass BinarySensorEntityDescription then add a logger.warning inside the constructor. But to be more specific, you could consider overriding the `is_on` function of the BinarySensorEntity, to raise a warning when someone is accessing the state of the deprecated sensor.

I am not sure this is the preferred approach but I think it is better to be targetted than have a deprecation warning raised on startup for every user of the integration.

Anonym-tsk commented:
> Please could you help provide a bit more context for this change?

I added "running state" sensor here https://github.com/home-assistant/core/pull/130444
But it was mistake. It is `Ignition state` in `run` field from starline api, not `running state`.
And i renamed this binary sensor in this PR.

But we already have "ignition state" in attributes of "engine" switch.
We discussed about it here https://github.com/home-assistant/core/pull/133566#issuecomment-2556768390 and decided to keep "ignition state" binary sensor but deprecate "ignition state" attribute of engine switch.

Anonym-tsk committed:
Merge branch 'dev' into starline_run_sensor_fix

joostlek commented on homeassistant/components/starline/account.py:
```diff
@@ -185,6 +185,7 @@ def engine_attrs(device: StarlineDevice) -> dict[str, Any]:
         """Attributes for engine switch."""
         return {
             "autostart": device.car_state.get("r_start"),
+            # Deprecated and should be removed in 2025.7
```
```suggestion
            # Deprecated and should be removed in 2025.8
```

joostlek committed:
Update homeassistant/components/starline/account.py

joostlek reviewed (changes requested):
Let's add a small description where we announce the deprecation

Anonym-tsk commented:
> Let's add a small description where we announce the deprecation

Where to add this? In docs?

Anonym-tsk committed:
Merge branch 'dev' into starline_run_sensor_fix

joostlek commented on homeassistant/components/starline/account.py:
```diff
@@ -185,6 +185,7 @@ def engine_attrs(device: StarlineDevice) -> dict[str, Any]:
         """Attributes for engine switch."""
         return {
             "autostart": device.car_state.get("r_start"),
```
So the other attribute also looks like it could be its own entity, would it make sense to also deprecate that at the same time so we can just tell the end user "all engine attributes" will be deprecated instead of 1 now and 1 in the future?

Anonym-tsk replied:
Done

joostlek reviewed (changes requested):
We can't detect deprecated state attributes. Changing the name also isnt a breaking change, since the unique id stays the same

Anonym-tsk committed:
Added autostart binary sensor

Anonym-tsk committed:
Merge branch 'dev' into starline_run_sensor_fix

joostlek commented:
Thanks for deprecating the other one as well!
